### PR TITLE
Coding Standards from Ticket 49222

### PR DESCRIPTION
**Pull request by mattyrob**
## Description
This PR includes all relevant changes included in upstream ticket 49222:
https://core.trac.wordpress.org/ticket/49222
Not all of the 8 commits were necessary or needed.
It also includes 45667 which renames `$r` to `$parsed_args` in several places for clarity.

## Motivation and context
Improve coding standards for readability and to aid move to PHP 8 and 8.1 compatibility.
It also commences work as described in the discussion on #1098 

## How has this been tested?
Coding standards and PHPUnit tests run and pass locally after these changes.

## Screenshots
N/A

## Types of changes
- Bug fix / Enhancement

### CODE REVIEW

**Comment by bahiirwa** [comment]
@mattyrob Conflict with `src/wp-admin/includes/image-edit.php`

File: src/wp-includes/user.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ function wp_signon( $credentials = array(), $secure_cookie = '' ) {
 	 *
 	 *     @type string $user_login    Username.
 	 *     @type string $user_password Password entered.
-	 *     @type bool   $remember      Whether to 'remember' the user. Increases the time
+	 *     @type bool   $parsed_argsemember      Whether to 'remember' the user. Increases the time
```
Comment (bahiirwa) [inline src/wp-includes/user.php:84]: Replacement of `$r` with `$parsed_args` beats the goal of having "good" variable names if the end result is `$parsed_argsember`

The changes in this file were not needed.

File: src/wp-includes/user.php [hunk 2]
```diff
@@ -421,9 +421,9 @@ function count_many_users_posts( $users, $post_type = 'post', $public_only = fal
 	$userlist = implode( ',', array_map( 'absint', $users ) );
 	$where    = get_posts_by_author_sql( $post_type, true, null, $public_only );
 
-	$result = $wpdb->get_results( "SELECT post_author, COUNT(*) FROM $wpdb->posts $where AND post_author IN ($userlist) GROUP BY post_author", ARRAY_N );
-	foreach ( $result as $row ) {
-		$count[ $row[0] ] = $row[1];
+	$parsed_argsesult = $wpdb->get_results( "SELECT post_author, COUNT(*) FROM $wpdb->posts $where AND post_author IN ($userlist) GROUP BY post_author", ARRAY_N );
```
Comment (bahiirwa) [inline src/wp-includes/user.php:424]: Similar comment as above.

File: src/wp-includes/user.php [hunk 3]
```diff
@@ -1026,7 +1026,7 @@ function setup_userdata( $for_user_id = '' ) {
  *
  * @since WP-2.3.0
  * @since WP-4.5.0 Added the 'display_name_with_login' value for 'show'.
- * @since WP-4.7.0 Added the `$role`, `$role__in`, and `$role__not_in` parameters.
+ * @since WP-4.7.0 Added the `$parsed_argsole`, `$parsed_argsole__in`, and `$parsed_argsole__not_in` parameters.
```
Comment (bahiirwa) [inline src/wp-includes/user.php:1029]: Same issue as above.

**Comment by bahiirwa** [comment]
Once we revert the changes in `src/wp-includes/user.php`. I would be happy to merge this. 

Replacement of `$r` with `$parsed_args` beats the goal of having "good" variable names if the end result is `$parsed_argsemember` for `$remember`

File: src/wp-includes/user.php (hunk 1 above)
Comment (mattyrob) [reply src/wp-includes/user.php:84]: I did something really stupid there - Search and Replace `$r` with `$parsed_args`!

I'll work on correcting my error.

**Review by bahiirwa — APPROVED** [review]
LGTM

**Comment by mattyrob** [comment]
> Once we revert the changes in `src/wp-includes/user.php`. I would be happy to merge this.
> 
> Replacement of `$r` with `$parsed_args` beats the goal of having "good" variable names if the end result is `$parsed_argsemember` for `$remember`

PR updated fixing the mistake highlighted above - thanks for spotting that @bahiirwa

**Review by xxsimoxx — APPROVED** [review]
Tested (together with #1102) on a live site with `WP_DEBUG` turned on, done common tasks and left in place for 3 days without any error or warning.

**Comment by mattyrob** [comment]
Discussed at weekly PR review and agreed for merge

**Pull request merged**